File: Problems/MeetingRooms/MeetingRooms.py
'''
Unlike Meeting Rooms II, we can just compare start time of current meeting with end time of previous meeting. If anyoverlap, return false.
'''

class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort(key=lambda x:x[0])
        
        for i in range(1, len(intervals)):
            if intervals[i][0] < intervals[i - 1][1]:
                return False
            
        return True
        

File: Problems/MeetingRooms/MeetingRooms.cpp
class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& x, vector<int>& y){return x[0] < y[0];});
        
        for(int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] < intervals[i - 1][1]) {
                return false;
            }
        }
        return true;
        
    }
};
